File: client/src/app/modules/tiers/services/competitor.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Competitor } from '../../tiers/model/Competitor';

@Injectable({
  providedIn: 'root'
})
export class CompetitorService extends AppRestService<Competitor>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getCompetitors() {
    return this.getListCached(new HttpParams(), "competitors");
  }

  addOrUpdateCompetitor(competitor: Competitor) {
    this.clearListCache(new HttpParams(), "competitors");
    return this.addOrUpdate(new HttpParams(), "competitor", competitor, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/tiers/service/CompetitorService.java
package com.jss.osiris.modules.tiers.service;

import java.util.List;

import com.jss.osiris.modules.tiers.model.Competitor;

public interface CompetitorService {
    public List<Competitor> getCompetitors();

    public Competitor getCompetitor(Integer id);
	
	 public Competitor addOrUpdateCompetitor(Competitor competitor);
}
